### PR TITLE
Code improvments

### PR DESCRIPTION
**Pull request by AntoinePrv**
Fix warnings on aggressive setting.

### CODE REVIEW

File: include/xwidgets/xbinary.hpp [hunk 1]
```diff
@@ -35,7 +36,7 @@ namespace xw
 
     XWIDGETS_API bool is_buffer_reference(const std::string& arg);
 
-    XWIDGETS_API int buffer_index(const std::string& v);
+    XWIDGETS_API std::size_t buffer_index(const std::string& v);
```
Comment (AntoinePrv) [inline include/xwidgets/xbinary.hpp:39]: Changing index type here.

File: include/xwidgets/xholder.hpp [hunk 2]
```diff
@@ -126,9 +126,6 @@ namespace xw
         public:
 
             xholder_impl() = default;
-            xholder_impl(xholder_impl&&) = delete;
-            xholder_impl& operator=(const xholder_impl&) = delete;
-            xholder_impl& operator=(xholder_impl&&) = delete;
```
Comment (AntoinePrv) [inline include/xwidgets/xholder.hpp:131]: Move semantics are defaulted in `xholder_id` that inherit from `xholder_impl`, making this invalid.

File: include/xwidgets/xcontroller.hpp [hunk 3]
```diff
@@ -116,7 +117,7 @@ namespace xw
 
         void set_defaults();
 
-        static int register_control_types();
```
Comment (AntoinePrv) [inline include/xwidgets/xcontroller.hpp:119]: The return value seems unused in Xwidget and part of private API.

File: src/xtarget.cpp [hunk 4]
```diff
@@ -12,55 +12,61 @@
...
 
-        xfactory& factory = get_xfactory();
-        factory.make(std::move(comm), state, buffers);
-    }
+            if (version.substr(0, version.find(".")) != XWIDGETS_STRINGIFY(XWIDGETS_PROTOCOL_VERSION_MAJOR))
+            {
+                throw std::runtime_error("Incompatible widget protocol versions");
+            }
 
-    int register_widget_target()
```
Comment (AntoinePrv) [inline src/xtarget.cpp:55]: Unused return value in an implementation detail.

**Comment by SylvainCorlay** [comment]
Nice, this looks good to me. Thanks!

**Pull request merged**